sunbird-rc-esignet-integration-impl/src/test/java/io/mosip/esignet/sunbirdrc/integration/service/SunbirdRCAuthenticaionServiceTest.java

```diff
@@ -0,0 +1,300 @@
+        ReflectionTestUtils.setField(sunbirdRCAuthenticationService, "fieldDetailList", fieldDetailList);
+        ReflectionTestUtils.setField(sunbirdRCAuthenticationService, "idField", "policyNumber2");
+        Throwable thrownException = Assert.assertThrows(Throwable.class,
+                () -> ReflectionTestUtils.invokeMethod(sunbirdRCAuthenticationService, "initialize"));
+
+
+        Assert.assertTrue(thrownException instanceof UndeclaredThrowableException);
+        Throwable actualException = ((UndeclaredThrowableException) thrownException).getUndeclaredThrowable();
+        Assert.assertTrue(actualException instanceof KycAuthException);
+        Assert.assertEquals("Invalid configuration: individual-id-field is not available in field-details.", actualException.getMessage());
+
+    }
+
+    @Test
+    public void doKycAuthWithValidParams_thenPass() throws KycAuthException, IOException, NoSuchFieldException, IllegalAccessException {
+        List<Map<String,String>> fieldDetailList = List.of(Map.of("id","policyNumber","type","string","format","string"),Map.of("id","fullName","type","string","format","string"));
+        ReflectionTestUtils.setField(sunbirdRCAuthenticationService, "fieldDetailList", fieldDetailList);
+        ReflectionTestUtils.setField(sunbirdRCAuthenticationService, "idField", "policyNumber");
+        ReflectionTestUtils.setField(sunbirdRCAuthenticationService, "registrySearchUrl", "url");
+        ReflectionTestUtils.setField(sunbirdRCAuthenticationService, "entityIdField", "policyNumber");;
+
+        // Arrange
+        String relyingPartyId = "validRelayingPartyId";
+        String clientId = "validClientId";
+        KycAuthDto kycAuthDto = new KycAuthDto(); // populate with valid data
+        AuthChallenge authChallenge=new AuthChallenge();
+        authChallenge.setFormat("string");
+        authChallenge.setAuthFactorType("KBA");
+        authChallenge.setChallenge("eyJmdWxsTmFtZSI6IlphaWQgU2lkZGlxdWUiLCJkb2IiOiIyMDAwLTA3LTI2In0=");
+
+        kycAuthDto.setChallengeList(List.of(authChallenge));
+        kycAuthDto.setIndividualId("000000");
+
+        List<Map<String,Object>> responseMap=new ArrayList<>();
+        Map<String,Object> map=Map.of("policyNumber","000000","dob","2000-07-26");
+        responseMap.add(map);
+        ResponseEntity<List<Map<String,Object>>>  responseEntity = new ResponseEntity(responseMap, HttpStatus.OK);
+        Mockito.when(restTemplate.exchange(
+                Mockito.any(RequestEntity.class),
+                Mockito.eq(new ParameterizedTypeReference<List<Map<String,Object>>>() {})
+        )).thenReturn(responseEntity);
+
+        Map<String,String> mockChallengMap=new HashMap<>();
+        mockChallengMap.put("fullName","Zaid Siddique");
+        mockChallengMap.put("dob","2000-07-26");
+        Mockito.when(objectMapper.readValue(Mockito.anyString(),Mockito.eq(Map.class))).thenReturn(mockChallengMap);
+
```

Comment:
https://github.com/mosip/digital-credential-plugins/pull/12#discussion_r1455988582 - Can you provide more details what is the issue with setting a object mapper new object via reflection on this test object ?

Reply:
When using reflection for object mapper getting this error 
"java.lang.NoClassDefFoundError:"

Reply:
to make use of reflection for objectmapper we need add this in our sunbird pom so that both at runtime and compile time same object get initilize:
		<dependency>
			<groupId>com.fasterxml.jackson.core</groupId>
			<artifactId>jackson-databind</artifactId>
			<version>2.11.3</version> <!-- Use the version that matches your project's requirements -->
		</dependency>

Reply:
There is a mismatch b/w  version 2.12.0 and 2.11.3

Reply:
so what to do ?

Reply:
mock or reflaction

Reply:
How is the runtime and compile time jackson versions are different ? Can we add this dependency only for test scope and move forward ?

Reply:
added scope as provided into pom of sunbird for jackson-databind